includes/Plugin_Context.php

```diff
@@ -183,4 +183,20 @@ public function minimum_supported_wp() {
 
 		return $this->minimum_supported_wp;
 	}
+
+	/**
+	 * Checks if the file is editable.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param string $file Filename.
+	 * @return bool true if the file is editable, otherwise false.
+	 */
+	public function is_file_editable( $file ) {
```

Comment:
The function still allowed edit link for `.eslintrc.js` and `.readme.txt` files while the plugin editor didn't allow it.

The plugin editor use `get_plugin_files` function to get the list of files in which it use `list_files` function that they [exclude the hidden files](https://github.com/WordPress/wordpress-develop/blob/6.5/src/wp-admin/includes/file.php#L161C4-L164C5). We should incorporate those changes so it will not include hidden files.

```
// Skip hidden and excluded files.
if ( ( ! $include_hidden && '.' === $file[0] ) || in_array( $file, $exclusions, true ) ) {
	continue;
}
```

**Plugin check:**
<img width="1557" alt="Screenshot 2024-07-15 at 10 00 18 AM" src="https://github.com/user-attachments/assets/a7524427-501c-45d4-bd65-4939fc5ac209">

**WP plugin editor:**
<img width="937" alt="Screenshot 2024-07-15 at 10 00 37 AM" src="https://github.com/user-attachments/assets/06170ec3-4572-4552-8280-da97ccd62702">

Reply:
PR updated.